src/workerd/io/hibernation-manager.c++

```diff
@@ -164,6 +185,39 @@ kj::Promise<void> HibernationManagerImpl::readLoop(HibernatableWebSocket& hib) {
   while (true) {
     kj::WebSocket::Message message = co_await ws.receive();
     // Note that errors are handled by the callee of `readLoop`, since we throw from `receive()`.
+
+    auto skip = false;
+
+    KJ_IF_MAYBE (reqResp, autoResponseRequestResponsePair) {
+      KJ_SWITCH_ONEOF(message) {
+        KJ_CASE_ONEOF(text, kj::String) {
+          if (text == (*reqResp)->getRequest()) {
+            // If the received message equals to the one set for auto response, we must
+            // short-circuit readLoop, store the current timestamp and respond with
+            // autoResponseResponse.
+            TimerChannel& timerChannel = KJ_REQUIRE_NONNULL(timer);
+            // We should have set the timerChannel previously in the hibernation manager.
+            // If we haven't, we aren't able to get the current time.
+            hib.autoResponseTimestamp = timerChannel.now();
+            KJ_IF_MAYBE(active, hib.activeOrPackage.tryGet<jsg::Ref<api::WebSocket>>()) {
+              // If the actor is not hibernated, we need to update autoResponseTimestamp on the
```

Comment:
/s/If the actor is not hibernated/If the _WebSocket is active_

The websocket could be inactive (hibernating) while the actor (and other websockets) are active.